cunumeric/deferred.py

```diff
@@ -1214,6 +1214,33 @@ def trilu(self, rhs, k, lower):
 
         task.execute()
 
+    # Repeat elements of an array.
+    @auto_convert([1])
```

Comment:
You probably want to remove this. `auto_convert` will always convert argument # 1 (i.e. `repeats`) to a `DeferredArray` before starting the function, and it looks like you want to do that conditionally. If I am reading this right then under the current code the scalar case in the C++ should never get triggered.

Reply:
Sorry, I'm wrong, the code may well be working as intended, because likely `task.add_scalar_arg(repeats, ty.int64)` can copy-out the scalar value from `DeferredArray`s. I think it would still be more appropriate to not go through `auto_convert` if you're dealing with a scalar.